examples/structured_2d_dgsem/elixir_euler_rayleigh_taylor_instability.jl

```diff
@@ -7,12 +7,66 @@ using Trixi
 gamma = 1.4
 equations = CompressibleEulerEquations2D(gamma)
 
+"""
+    initial_condition_rayleigh_taylor_instability(coordinates, t, equations::CompressibleEulerEquations2D)
+
+Setup used for the Rayleigh-Taylor instability. Initial condition adapted from
+- Shi, Jing, Yong-Tao Zhang, and Chi-Wang Shu (2003).
+  Resolution of high order WENO schemes for complicated flow structures.
+  [DOI](https://doi.org/10.1016/S0021-9991(03)00094-9).
+
+- Remacle, Jean-François, Joseph E. Flaherty, and Mark S. Shephard (2003).
+  An adaptive discontinuous Galerkin technique with an orthogonal basis applied to compressible
+  flow problems.
+  [DOI](https://doi.org/10.1137/S00361445023830)
+
+The domain is [0, 0.25] x [0, 1]. Boundary conditions can be reflective wall boundary conditions on
+all boundaries or
+- periodic boundary conditions on the left/right boundaries
+- Dirichlet boundary conditions on the top/bottom boundaries
+
+This should be used together with `source_terms_rayleigh_taylor_instability`, which is
+defined below.
+"""
+@inline function initial_condition_rayleigh_taylor_instability(x, t,
+                                                               equations::CompressibleEulerEquations2D,
+                                                               slope=1000)
+  tol = 1e2*eps()
+
+  if x[2] < .5
+    p = 2*x[2] + 1
+  else
+    p = x[2] + 3/2
+  end
+
+  # smooth the discontinuity to avoid ambiguity at element interfaces
+  smoothed_heaviside(x, left, right) = left + 0.5*(1 + tanh(slope * x)) * (right-left)
+  rho = smoothed_heaviside(x[2] - 0.5, 2.0, 1.0)
+
+  c = sqrt(equations.gamma * p / rho)
+  # the velocity is multiplied by sin(pi*y)^6 as in Remacle et al. 2003 to ensure that the
+  # initial condition satisfies reflective boundary conditions at the top/bottom boundaries.
+  v = -0.025 * c * cos(8*pi*x[1]) * sin(pi*x[2])^6
+  u = 0.0
+
+  return prim2cons(SVector(rho, u, v, p), equations)
+end
+
+@inline function source_terms_rayleigh_taylor_instability(u, x, t,
+                                                          equations::CompressibleEulerEquations2D)
+  g = 1.0
+  rho, rho_v1, rho_v2, rho_e = u
+
+  return SVector(0.0, 0.0, g*rho, g*rho_v2)
+end
+
+surface_flux = flux_hll
```

Comment:
```suggestion
```
I think your previous version is fine - but that's bikeshedding, so feel free to ignore this comment and the next if you don't like it.